src/main/java/gregtech/api/util/GTHashMaps.java

```diff
@@ -25,17 +25,19 @@ public class GTHashMaps {
      * @return a {@link Map} of {@link ItemStackKey} and {@link Integer} as amount on the inventory
      */
     public static Map<ItemStackKey, Integer> fromItemHandler(IItemHandler inputs) {
-        final Supplier<Map<ItemStackKey, Integer>> mapSupplier = Object2IntLinkedOpenHashMap::new;
+        final Map<ItemStackKey, Integer> map = new Object2IntLinkedOpenHashMap<>();
```

Comment:
Is there any reason why we're still using these keys instead of providing a hash strategy straight into a fastutil custom hash map?

Reply:
Unfortunately yes, changing the map to a <ItemStack, Integer> means fixing some of parallel logic and reworking the item merging logic.

Reply:
This codebase is cursed